nodeadm/internal/system/local_disk.go

```diff
@@ -0,0 +1,35 @@
+package system
+
+import (
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/awslabs/amazon-eks-ami/nodeadm/internal/api"
+	"go.uber.org/zap"
+)
+
+const localDiskAspectName = "local-disk"
+
+func NewLocalDiskAspect() SystemAspect {
+	return &localDiskAspect{}
+}
+
+type localDiskAspect struct{}
+
+func (a *localDiskAspect) Name() string {
+	return localDiskAspectName
+}
+
+func (a *localDiskAspect) Setup(cfg *api.NodeConfig) error {
+	if cfg.Spec.Instance.LocalStorage.Strategy == "" {
+		zap.L().Info("Not configuring local disks!")
+		return nil
+	}
+	strategy := strings.ToLower(string(cfg.Spec.Instance.LocalStorage.Strategy))
+	// #nosec G204 Subprocess launched with variable
+	cmd := exec.Command("setup-local-disks", strategy)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
```

Comment:
its nice having this in the nodeadm log, just weird that it infringes on the json logging we "kind" of have

Reply:
Yeah, I could capture this and log it with zap, but it'd be less readable IMO and at this point we aren't actually ingesting the structured logging, it's just a way to keep the code clean with fields vs format strings.

Reply:
totally agree 👍